code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -224,6 +224,13 @@
 	spare.domutcheck()
 	spare.Move(get_step(H.loc, pick(NORTH,SOUTH,EAST,WEST)))
 
+	if(owner.GetComponent(/datum/component/nanites))

```

Comment:
You can do this whole thing without GetComponent. Simply just add a signal that nanites are registered to and call it COMSIG_GET_NANITE_VOLUME. Send down a list with that signal just add the amount in the listening proc.
Basically:
```
var/list/L = list()
SEND_SIGNAL(owner, COMSIG_GET_NANITE_VOLUME, L)
var/new_amount = L[1] * 0.5
if(new_amount)
	SEND_SIGNAL(owner, COMSIG_NANITE_SET_VOLUME, amount)
	spare.AddComponent(/datum/component/nanites, new_amount)
	SEND_SIGNAL(spare, COMSIG_NANITE_SYNC, owner)
```
and you also need to add/change some stuff in the nanite thingy.
```
//Syncs the nanite component to another, making it so programs are the same with the same programming (except activation status)
/datum/component/nanites/proc/sync(datum/signal_source, mob/source, full_overwrite = TRUE, copy_activation = FALSE)	
	var/list/programs_to_add = list()
	SEND_SIGNAL(source, COMSIG_NANITE_GET_PROGRAMS, programs_to_add)
	if(!LAZYLEN(programs_to_add)
		return
	var/list/programs_to_remove = programs.Copy()
	for(var/X in programs)
		var/datum/nanite_program/NP = X
		for(var/Y in programs_to_add)
			var/datum/nanite_program/SNP = Y
			if(NP.type == SNP.type)
				programs_to_remove -= NP
				programs_to_add -= SNP
				SNP.copy_programming(NP, copy_activation)
				break
	if(full_overwrite)
		for(var/X in programs_to_remove)
			qdel(X)
	for(var/X in programs_to_add)
		var/datum/nanite_program/SNP = X
		add_program(null, SNP.copy())
```
this will also need 2 line changed in utility.dm and an updated commend in _DEFINES/components.dm

and of course the proc that listens to
```
/datum/component/nanites/proc/signal_get_volume(datum/source, list/L)
	L[1] += nanite_volume
```